scripts/jest/jest-cli.js

```diff
@@ -0,0 +1,268 @@
+  devtools: './scripts/jest/config.build-devtools.js',
+};
+
+// TODO: These configs are separate but should be rolled into the configs above
+// so that the CLI can provide them as options for any of the configs.
+const persistentConfig = './scripts/jest/config.source-persistent.js';
+const buildConfig = './scripts/jest/config.build.js';
+
+const argv = yargs
+  .parserConfiguration({
+    // Important: This option tells yargs to move all other options not
+    // specified here into the `_` key. We use this to send all of the
+    // Jest options that we don't use through to Jest (like --watch).
+    'unknown-options-as-args': true,
+  })
+  .wrap(yargs.terminalWidth())
+  .options({
+    debug: {
+      alias: 'd',
+      describe: 'Run with node debugger attached.',
+      requiresArg: false,
+      type: 'boolean',
+      default: false,
+    },
+    project: {
+      alias: 'p',
+      describe: 'Run the given project.',
+      requiresArg: false,
+      type: 'string',
+      default: 'oss',
+      choices: Object.keys(configMap),
+    },
+    releaseChannel: {
+      alias: 'r',
+      describe: 'Run with the given release channel.',
+      requiresArg: false,
+      type: 'string',
+      default: 'experimental',
+      choices: ['experimental', 'stable'],
+    },
+    env: {
+      alias: 'e',
+      describe: 'Run with the given node environment.',
+      requiresArg: false,
+      type: 'string',
+      choices: ['development', 'production'],
+    },
+    prod: {
+      describe: 'Run with NODE_ENV=production.',
+      requiresArg: false,
+      type: 'boolean',
+      default: false,
+    },
+    dev: {
+      describe: 'Run with NODE_ENV=development.',
+      requiresArg: false,
+      type: 'boolean',
+      default: false,
+    },
+    variant: {
+      alias: 'v',
+      describe: 'Run with www variant set to true.',
+      requiresArg: false,
+      type: 'boolean',
+      default: false,
+    },
+    build: {
+      alias: 'b',
+      describe: 'Run tests on builds.',
+      requiresArg: false,
+      type: 'boolean',
+      default: false,
+    },
+    persistent: {
+      alias: 'n',
+      describe: 'Run with persistence.',
+      requiresArg: false,
+      type: 'boolean',
+      default: false,
+    },
+    ci: {
+      describe: 'Run tests in CI',
+      requiresArg: false,
+      type: 'boolean',
+      default: false,
+    },
+  }).argv;
+
+function logError(message) {
+  console.error(chalk.red(`\n${message}`));
+}
+function validateOptions() {
+  let success = true;
+  if (argv.variant && argv.project !== 'www') {
+    logError(
+      'Variant is only supported for the www config. Update these options to continue.'
+    );
+    success = false;
+  }
+
+  if (argv.prod && argv.project === 'devtools') {
+    logError('DevTools do not support --prod. Remove this option to continue.');
+    success = false;
+  }
+
+  if (argv.env === 'production' && argv.project === 'devtools') {
```

Comment:
DevTools doesn't support any of the following flags: `--dev`, `--prod`, `--env`, `--variant`, `--persistent `

Really it only supports `--debug` (and I guess `--build`) for the time being.

nit: Might be a little easier to read and reason about this if you grouped all of the "if devtools" conditionals within a single block, e.g.

```js
if (project === 'devtools') {
  // Other checks here...
}
```

Reply:
I'll update - can the devtools run without `--build`?

Reply:
No. It's kind of maybe unfortunate, but DevTools (including its tests) can only run against a pre-built version of React for the time being.

Reply:
The reason I said "and I guess `--build`" is because it isn't really a configurable option. Only one value is allowed 😅

Reply:
Oh I see, so users should run `yarn build` before running the devtools tests? So we'll want to include the build directory check (I was excluding it). Will update now 👍

Reply:
In most cases they can just run `./scripts/release/download-experimental-build.js`.

The only reason to run `yarn build-for-devtools` is to test a local change.